### ISSUE TITLE
vold via Magisk delete all app data

### ISSUE THREAD

**Issue by GlitchedCod** [noname8964/vold-posix#2]
Hello,
After the installation of the Vold module created by noname8964, my phone have deleted all app data and set the memory of the phone as "read only". My phone is a Xiaomi Redmi Note 5 Official MIUI 9 ROM unlocked.

Please solve this error. 😢

Thank you.

**Comment by noname8964** [comment]
I said it only works for AOSP-like system:( MIUI or other UI rom has not tested.
and in my opinion, MIUI support exfat, why you use this module?

**Comment by GlitchedCod** [comment]
but not NTFS, i have just installed that in the case of if i wanted to watch a movie in a hard drive or something like that...
I have see that i have flash the RU ROM (so the bad one) so I will take the opportunity to change to the original ROM now...

**Comment by kalawy** [comment]
i am no expert but aaddind the libs and bin of ntfs 
integrated successfully and worked perfectly on a rom that had vold for exfat but not ntfs 

ypu may give it a try 

 https://forum.xda-developers.com/mi-max-2/development/stable-msm-xtended-v2-5-t3805375/post78002283

**Comment by noname8964** [comment]
> i am no expert but adding the libs and bin of ntfs
> integrated successfully and worked perfectly on a rom that had vold for exfat but not ntfs
> 
> you may give it a try
> 
> https://forum.xda-developers.com/mi-max-2/development/stable-msm-xtended-v2-5-t3805375/post78002283

Nope, he's on official MIUI that didnt support these vold binary or fuse libraries. Since MIUI is close source, I won't plan to support miui or other custom ui rom.

The v1 version will occurs weird issue, but v2 mostly fixes this. maybe v2 can work with miui but I can't promise anything

**Comment by noname8964** [comment]
@diletantEG Once again, this module didn't support non AOSP-like system. All bugs with these system won'r fix.
And encryption isn't the issue. my Nokia phone works fine with encrypt user data. Maybe MIUI has some custom methods in there boot image, but I can't fix this.
If you really need ntfs driver, maybe you can try [Paragon driver](https://play.google.com/store/apps/details?id=com.paragon.tcplugins_ntfs_ro) from Google Play or flash LineageOS/etc.

**Comment by GlitchedCod** [comment]
@diletantEG maybe this have a relationship with the ROM of the phone, when i have installed the latest version of vold, i used MIUI 9.2 Russian model

**Comment by diletantEG** [comment]
I have two absolutely identical phones, with the same system (miui 10.0.3.0) and a similar set of software. This is surprising, why everything was installed perfectly on one thing, and this problem arose on the other.

**Comment by blaueente** [comment]
Same problem here, with BQ Aquaris X2 pro (zangyapro), on firmware 1.5.0: device reports defect encryption, and factory-resets.
What does  "AOSP-like" mean in this context?
It is a "Android One" device, so it should not have few vendor-specific adjustments.
BQ shamingly does not ship exfat/ntfs drivers, so having this module would be really nice.
Is there anything I can help to debug?

**Comment by noname8964** [comment]
Android one should work. past the logcat. use ``adb logcat|grep -i vold``
to get the logcat. maybe also need ``adb logcat|grep avc:``
PS: in Windows you can use ``find`` instead of ``grep -i``, but idk if
``find`` is case-sensitive or not. it need ignore case.

blaueente <notifications@github.com> 于 2018年11月17日周六 21:18写道：

**Comment by blaueente** [comment]
When should I get the logs? During normal runtime, without the vold module? During Module installation? During the failed boot, but will this work at all? Will it help to make a backup of some partition such that I can try restore the phone without wiping and re-installing? There's no TWRP available, but i can use fastboot.

**Comment by noname8964** [comment]
nope, this only get the log that I can find something in it. run it on boot
up. but if you don't have TWRP at all, I don't recommend you use this
module until I find the problem. you can flash your boot image backup to
restore what this module changed. iirc when decrypt failed it will warn you
to do a factory reset but if you don't touch anything it won't do anything.
BTW, what's your phone's status atm?

blaueente <notifications@github.com> 于 2018年11月17日周六 21:50写道：

**Comment by blaueente** [comment]
Thanks for your help!
My phone is working again, after a the factory reset, magisk is still on there.
I'll now just re-format my sdcard for now, and wait for TWRP being available.
Just for theory, maybe I want to experiment later; if I do the following:

1. pull a boot image backup
2. install vold module 
3. reboot, (immediately?) start adb logcat | tee debug.log
4. get warning, don't touch anything
5. reboot into bootloader, re-flash bootloader backup

Do you think step 5 will work without factory-reset? I understand, its my risk anyway ;-)

**Comment by dariods8474** [comment]
> I said it only works for AOSP-like system:( MIUI or other UI rom has not tested.
> and in my opinion, MIUI support exfat, why you use this module?

Most AOSP roms have NTFS support than what is the use of this module? Module for stock Miui roms have exfat support where the file integrity is doubtful which arises the Need for ext or NTFS support for SD card

**Comment by null4n** [comment]
> Most AOSP roms have NTFS support than what is the use of this module? Module for stock Miui roms have exfat support where the file integrity is doubtful which arises the Need for ext or NTFS support for SD card

stock pixel and some stock aosp-like system didn't support exfat and etc. I made this module only for these systems. miui has lots of changes from aosp, so I won't support it. you can still use it in your risk, but don't tell me any bug or losing data

**Comment by dariods8474** [comment]
> > Most AOSP roms have NTFS support than what is the use of this module? Module for stock Miui roms have exfat support where the file integrity is doubtful which arises the Need for ext or NTFS support for SD card
> 
> stock pixel and some stock aosp-like system didn't support exfat and etc. I made this module only for these systems. miui has lots of changes from aosp, so I won't support it. you can still use it in your risk, but don't tell me any bug or losing data

I wish you could make a universal module. You are the only one who has attempted. My bad luck not to have one.

**Comment by null4n** [comment]
> I wish you could make a universal module. You are the only one who has attempted. My bad luck not to have one.

I don't think so, vold has lots of changes between different api level. the only way maybe compile all of theme, but this take too many time and disk space. amd I need different system to test it.  unfortunately I only have 8.1
